variables.tf

```diff
@@ -32,6 +32,37 @@ variable "karpenter" {
   }
 }
 
+variable "enable_datadog" {
+  description = "Enable Datadog integration"
+  type        = bool
+  default     = false
+}
+
+variable "datadog" {
+  description = "Map of Datadog configurations"
+  type        = any
+  default     = {}
+  # TODO: currently possible values. How to document them?
+  # agent_manifest_override
+  # agent_manifest_features
+  # agent_api_key_name
+  # agent_app_key_name
+  # operator_chart_version
+}
```

Comment:
How do we document sub-keys used in the code?

Reply:
Also an issue that recently arose is that with variables that are of type any we can incur in mistakes. Currently the examples in this repo set the vpc.subnets_config object but this is not currently in use in the k8s platform. With variables properly set we would have spot this issue sooner, therefore I am thinking of re-writing this with all the fields and the types, including the terraform optionals.